Source/WebCore/css/SelectorChecker.cpp

```diff
@@ -1271,9 +1271,22 @@ bool SelectorChecker::checkOne(CheckingContext& checkingContext, LocalContext& c
             if (checkingContext.pseudoId != CSSSelector::pseudoId(selector.pseudoElement()) || !selector.argumentList())
                 return false;
 
-            // Wildcard always matches.
+            // Universal selector always matches.
             auto& argument = selector.argumentList()->first();
-            return argument == starAtom() || argument == checkingContext.pseudoElementNameArgument;
+            if (argument != starAtom() && argument != checkingContext.pseudoElementNameArgument)
+                return false;
+
+            if (selector.argumentList()->size() == 1)
+                return true;
+
+            auto result = std::ranges::all_of(
+                selector.argumentList()->begin() + 1, selector.argumentList()->end(),
```

Comment:
Would read so much more nicely with the local variable and less wrapping:
```
auto result = std::ranges::all_of(list.begin() + 1, list.end(),
```

Reply:
Good suggestion!